### ISSUE TITLE
ProcessTest on Apple Mx hardware should test `emulated` 

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2638]
Using Apple M1 hardware & ARM javalib (zulu)

I expect the tests below to succeed with an ARM javalib
and they do not. I understand why they might be
disabled with an X86_64 emulated javalib.

What disturbs my tranquility is that tracking this down
for the 99th time increases the time I spend testing
fixes on M1 for real bugs. 

When I get rid of all the bees in my bonnet, where are
the bees going to live?

(When setting up an Apple Mx system it is all too easy
to fall into the hidden punji pit of using an emulated
javalib, but that is another Issue. I will try to report
it when I have reduced my queue of annoyances.) 

Using Apple M1 hardware & ARM javalib (zulu)
```
[info] Test javalib.lang.ProcessTest.destroy started
[warn] Test assumption in test javalib.lang.ProcessTest.destroy failed: org.junit.AssumptionViolatedException: Breaks test runner when executed in emulator, took 0.001 sec

[info] Test javalib.lang.ProcessTest.destroyForcibly started
[warn] Test assumption in test javalib.lang.ProcessTest.destroyForcibly failed: org.junit.AssumptionViolatedException: Breaks test runner when executed in emulator, took 0.000 sec

```
A hypothetical `isAppleEmulated()` test should be pretty easy. I have the code, just not
integrated for a PR.

**Comment by ekrich** [comment]
Even though I have an M1, I have not chased bees either. There are a lot of tests not running and we have used the term "blacklisted" which is not good either. Theoretically the Process test should work on M1 either native or Rosetta but that may take a bit of `#ifdef` wizardry.

**Comment by LeeTibbert** [comment]
Thank you for the info.

From the comments, I think Rosetta has a problem handling signals/exceptions and
they blow it up.  That might be an un-earned slur, but I thin I experienced it when
I commented out the apparently wrong ifMacMumble condition. Only spent a week
or so figuring out that I was having problems because I was using the 'obvious'
Eclipse JVM, which runs under Rosetta.   I think a comment from you in 
one of your many helpful writings lead me to the Azul Zulu native JVM. 
A big THANK YOU. 

re: "blacklisted"  I concur, we can do better in informal discussions and code
being written.
